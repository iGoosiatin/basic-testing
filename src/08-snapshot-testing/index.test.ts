import { generateLinkedList } from './index';

describe('generateLinkedList', () => {
  // Check match by expect(...).toStrictEqual(...)
  test('should generate linked list from values 1', () => {
    const linkedList = generateLinkedList([1, 2]);
    expect(linkedList).toStrictEqual({
      next: {
        next: {
          next: null,
          value: null,
        },
        value: 2,
      },
      value: 1,
    });
  });

  // Check match by comparison with snapshot
  test('should generate linked list from values 2', () => {
    const linkedList = generateLinkedList([1, 2, 3, 4]);
    expect(linkedList).toMatchSnapshot();
  });
});
