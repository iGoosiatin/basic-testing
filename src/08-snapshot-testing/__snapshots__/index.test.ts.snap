// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateLinkedList should generate linked list from values 2 1`] = `
Object {
  "next": Object {
    "next": Object {
      "next": Object {
        "next": Object {
          "next": null,
          "value": null,
        },
        "value": 4,
      },
      "value": 3,
    },
    "value": 2,
  },
  "value": 1,
}
`;
